lib/Header.js

```diff
@@ -1,37 +1,90 @@
-import React from "react";
-import PropTypes from "prop-types";
-import { Text, utils, widgetCreator } from "@bpanel/bpanel-ui";
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text, utils, widgetCreator } from '@bpanel/bpanel-ui';
 
 const { connectTheme } = utils;
 
-const Header = ({ statusIcon, network, bcoinUri, theme }) => (
-  <div className="container">
-    <div
-      className={`${theme.headerbar.networkStatus} ml-md-auto text-right col`}
-    >
-      <div className="network text-uppercase">
-        <Text className={theme.headerbar.text}>
-          Status: {network}
-          <i
-            className={`${theme.headerbar.icon} fa fa-${statusIcon}`}
-            areahidden="true"
-          />
-        </Text>
-      </div>
-      <div className="node">
-        <Text className={`${theme.headerbar.nodeText} ${theme.headerbar.text}`}>
-          Node:{" "}
-        </Text>
-        <Text className={`${theme.headerbar.text}`}>{bcoinUri}</Text>
+class Header extends React.PureComponent {
+  constructor(props) {
+    super(props);
+  }
+
+  static get propTypes() {
+    return {
+      theme: PropTypes.object,
+      network: PropTypes.string,
+      setCurrentClient: PropTypes.func.isRequired,
+      statusIcon: PropTypes.string,
+      clients: PropTypes.object
+    };
+  }
+
+  onConfigChange(e) {
+    const { clients: { clients }, setCurrentClient } = this.props;
+    const id = e.target.value;
+    if (!clients[id]) console.error(`Client with id ${id} does not exist`);
+    else setCurrentClient({ id, ...clients[id] });
+  }
+
+  render() {
+    const {
+      theme,
+      statusIcon,
+      network,
+      clients,
+      setCurrentClient
+    } = this.props;
+    let nodeId = (
```

Comment:
Maybe creating a class method that creates the `nodeId` and invoking it would make the render function smaller